convert2rhel/utils.py

```diff
@@ -453,4 +454,18 @@ def backup(self):
             loggerinst.warning("Can't access %s" % TMP_DIR)
 
 
+def parse_version_number(version_str=None):
+    if version_str:
+        try:
+            version_full_number = re.search(r"([\d.]+)", version_str).group()
+            regex = re.search(r"^(\d+)?\.?(\d+)?\.?(\*|\d+)?$", version_full_number)
+            res = {'major': regex.group(1) if regex.group(1) else None,
+                   'minor': regex.group(2) if regex.group(2) else None,
+                   'release': regex.group(3) if regex.group(3) else None}
+
+            return res
+        except:
+            return
```

Comment:
Try something like this instead. I'll explain below
```suggestion
def parse_version_number(version_str):
    """
    Parses the version number from the version string in /etc/system-release.
    Returns a dict with major, minor, and release keys where each key could be None
    """
    if not isinstance(version_str, str) :
        raise TypeError("Expected version_str to be a string")
    version_number_regex = re.search(r"(^|\s)(\d+)?\.?(\d+)?\.?(\*|\d+)", version_str)
    if version_number_regex is None:
    	raise Exception("Could not find Operating System version number in \"" + version_str + "\"")
    split_version = version_number_regex.groups()
    version_dict = {
        'major': split_version[0],
        'minor': split_version[1],
        'release': split_version[2]
    } 
    
    return version_dict
```

We're parsing something here so we don't want the parameter to be optional, then what's the point of the function? So we can remove that and make it a required argument. 

Instead if checking if it's "true", it's better to check if it's a string, so here we do a `not string` check in the beginning to reduce indentation.

RegEx can be simplified to one RegEx. Doing multiple RegEx doesn't serve much purpose. We can also change the RegEx slightly.

As for groups, doing `if re.group(index)` will always raise `IndexError: no such group` if there is no group, but `re.groups()` will default empty groups to None. Meaning, as long as we have 3 groups in the RegEx pattern we will get a `None` in the tuple

The error raised is simply to have a nicer error if we don't find any matches in the string. We should however use our own Exception here and not the base Exception